Source/Core/Core/AchievementManager.cpp

```diff
@@ -152,66 +158,218 @@ void AchievementManager::LoadGameByFilenameAsync(const std::string& iso_path,
           },
       .close = [](void* file_handle) { delete reinterpret_cast<FilereaderState*>(file_handle); }};
   rc_hash_init_custom_filereader(&volume_reader);
-  if (!rc_hash_generate_from_file(m_game_hash.data(), RC_CONSOLE_GAMECUBE, iso_path.c_str()))
+  m_queue.EmplaceItem([this, callback, file_path] {
+    Hash new_hash;
+    if (!rc_hash_generate_from_file(new_hash.data(), RC_CONSOLE_GAMECUBE, file_path.c_str()))
+    {
+      ERROR_LOG_FMT(ACHIEVEMENTS, "Unable to generate achievement hash from game file {}.",
+                    file_path);
+      callback(AchievementManager::ResponseType::MALFORMED_OBJECT);
+    }
+    {
+      std::lock_guard lg{m_lock};
+      m_game_hash = std::move(new_hash);
```

Comment:
You may want to check here if the achievement manager has been disabled in the meantime.